src/Action/Action.jl

```diff
@@ -386,34 +379,56 @@ function scan(
 
     # determine current mesh layout
     num_lin = ceil(Int64, p0 * (length(x) - 1))
-    δ       = num_lin * x[2]
 
-    # determine position of the maximum
-    max_idx = argmax(abs.(y))
+    # a) check for peaks at finite frequencies and set δa as 'point where channel has decayed to 10 percent of peak value'
+    δa  = Inf
+    idx = argmax(abs.(y))
+
+    if idx > 1 
+        for i in idx + 1 : length(x)
+            if abs(y[i] / y[idx]) <= 0.1
+                δa = x[i]
+                break
+            end 
+        end 
+    end
 
-    # if the maximum is not located at the origin, set linear extent via position of the maximum
-    if max_idx > 1
-        δ = 1.2 * x[max_idx]
-    # else set linear extent such that p1 <= Δ <= p2, where Δ is the relative deviation between the value at the origin and the first finite frequency
-    else
-        # determine relative deviation from origin to first finite frequency
+    # b) check for sign change at finite frequencies and set δb accordingly 
+    δb = Inf
+
+    for i in 2 : length(y)
+        if sign(y[i]) != sign(y[1]) 
+            δb = x[i]
+            break
+        end 
+    end
```

Comment:
I think this should not happen, since we only scan around the maximum of the channels, and even if this happens on a rare occasion, the minimum step size provides a lower bound to cope with that.

Reply:
Agreed with @dominikkiese